src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,29 @@
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = factory.openSession();
+        if (session == null) {
+            throw new DataProcessingException("Session not initialized");
+        }
+        Transaction transaction = session.beginTransaction();
```

Comment:
Use try-with-resources or try-finally to ensure that the session is closed even if an exception occurs. This will make the code more robust and prevent potential resource leaks.